### DISCUSSION TITLE
How can i keep a global instance which i can't be dropped during the application running

### DISCUSSION THREAD

**Discussion by jinkaz** [rwf2/Rocket#2571] [Questions]
i wanner integrate a sdk which connected a service like consul as its client.The sdk should be needed to keep it globally during my application running lifecyle. i prefer to use state to do it ,but i not sure that is a good idea just like below code:
#[launch]
fn rocket() -> _ {
    let a = Arc::new(Sm::init());
    let sm = Arc::clone(&a);

    rocket::build().manage(sm)
}

pub struct Sm;

impl Sm {
    pub  fn init()->impl NamingService{
     //.....
}}

thanks in advance

**Comment by zolagonano** [comment]
That actually seems like a good idea but there are other ways to achieve that, for example by using `lazy_static` crate and `Mutex`.

```rust
lazy_static! {
    static ref SM: Arc<Mutex<Sm>> = Arc::new(Mutex::new(Sm::init()));
}
```
